src/main/java/uk/gov/ons/fwmt/job_service/rest/impl/FieldPeriodResourceServiceImpl.java

```diff
@@ -31,16 +28,13 @@ public FieldPeriodResourceServiceImpl(
 
   @Override
   public Optional<FieldPeriodDto> findByFieldPeriod(final String fieldPeriod) {
-    log.info("findByFieldPeriod: {}", fieldPeriod);
-    try {
-      final ResponseEntity<FieldPeriodDto> responseEntity = restTemplate
-          .getForEntity(findUrl, FieldPeriodDto.class, fieldPeriod);
-      if (responseEntity.getStatusCode().equals(HttpStatus.OK)) {
-        return Optional.ofNullable(responseEntity.getBody());
-      }
-    } catch (HttpClientErrorException httpException) {
-      log.error("An error occurred while communicating with the resource service", httpException);
+    log.debug("findByFieldPeriod entered: fieldPeriod={}", fieldPeriod);
+    final Optional<FieldPeriodDto> jobDto = RestHelper.get(restTemplate, findUrl, FieldPeriodDto.class, fieldPeriod);
+    if (jobDto.isPresent()) {
```

Comment:
no functional checks to just log.
Log in the caller class based on there usage.